src/base1/Makefile.am

```diff
@@ -23,6 +23,8 @@ basedebug_DATA = \
 	dist/base1/cockpit.min.js.map \
 	$(NULL)
 
+AM_TESTS_ENVIRONMENT = export G_DEBUG=fatal-criticals ;
```

Comment:
I actually did consider/try that, but a lot of our tests (rightfully) cover error conditions which trigger warnings (11 failures, 3 errors).

Arguably we should mark them somehow and also avoid actually printing expected warnings to the console (as they look unclean and are a bit confusing), but that's a lot more work and IMHO subject for a separate PR.